src/decompress.c

```diff
@@ -0,0 +1,662 @@
+        mov r0, r9\n\
+        cmp r0, 0x7\n\
+        bne _0800EF7C\n\
+        movs r1, 0x80\n\
+        lsls r1, 1\n\
+        adds r5, r1\n\
+        b _0800EF88\n\
+    _0800EF7C:\n\
+        mov r2, r9\n\
+        cmp r2, 0x5\n\
+        bne _0800EF88\n\
+        movs r4, 0xF0\n\
+        lsls r4, 1\n\
+        adds r5, r4\n\
+    _0800EF88:\n\
+        ldr r0, [sp, 0x14]\n\
+        ldr r1, [sp]\n\
+        cmp r0, r1\n\
+        bge _0800EF92\n\
+        b _0800EE16\n\
+    _0800EF92:\n\
+        b _0800F024\n\
+    _0800EF94:\n\
+        movs r6, 0\n\
+        ldr r2, [sp]\n\
+        cmp r6, r2\n\
+        bge _0800F024\n\
+    _0800EF9C:\n\
+        adds r6, 0x1\n\
+        mov r10, r6\n\
+        mov r4, r9\n\
+        cmp r4, 0x6\n\
+        bne _0800EFB4\n\
+        movs r0, 0\n\
+        movs r3, 0xFF\n\
+    _0800EFAA:\n\
+        strb r0, [r5]\n\
+        adds r5, 0x1\n\
+        subs r3, 0x1\n\
+        cmp r3, 0\n\
+        bge _0800EFAA\n\
+    _0800EFB4:\n\
+        movs r7, 0\n\
+        cmp r7, r9\n\
+        bge _0800F008\n\
+    _0800EFBA:\n\
+        adds r7, 0x1\n\
+        mov r8, r7\n\
+        mov r1, r9\n\
+        lsls r0, r1, 5\n\
+        cmp r1, 0x6\n\
+        bne _0800EFD4\n\
+        movs r1, 0\n\
+        movs r3, 0x1F\n\
+    _0800EFCA:\n\
+        strb r1, [r5]\n\
+        adds r5, 0x1\n\
+        subs r3, 0x1\n\
+        cmp r3, 0\n\
+        bge _0800EFCA\n\
+    _0800EFD4:\n\
+        adds r1, r0, 0\n\
+        cmp r1, 0\n\
+        ble _0800EFEE\n\
+        adds r3, r1, 0\n\
+    _0800EFDC:\n\
+        ldr r2, [sp, 0x4]\n\
+        ldrb r0, [r2]\n\
+        strb r0, [r5]\n\
+        adds r2, 0x1\n\
+        str r2, [sp, 0x4]\n\
+        adds r5, 0x1\n\
+        subs r3, 0x1\n\
+        cmp r3, 0\n\
+        bne _0800EFDC\n\
+    _0800EFEE:\n\
+        mov r4, r9\n\
+        cmp r4, 0x6\n\
+        bne _0800F002\n\
+        movs r0, 0\n\
+        movs r3, 0x1F\n\
+    _0800EFF8:\n\
+        strb r0, [r5]\n\
+        adds r5, 0x1\n\
+        subs r3, 0x1\n\
+        cmp r3, 0\n\
+        bge _0800EFF8\n\
+    _0800F002:\n\
+        mov r7, r8\n\
+        cmp r7, r9\n\
+        blt _0800EFBA\n\
+    _0800F008:\n\
+        mov r0, r9\n\
+        cmp r0, 0x6\n\
+        bne _0800F01C\n\
+        movs r0, 0\n\
+        movs r3, 0xFF\n\
+    _0800F012:\n\
+        strb r0, [r5]\n\
+        adds r5, 0x1\n\
+        subs r3, 0x1\n\
+        cmp r3, 0\n\
+        bge _0800F012\n\
+    _0800F01C:\n\
+        mov r6, r10\n\
+        ldr r1, [sp]\n\
+        cmp r6, r1\n\
+        blt _0800EF9C\n\
+    _0800F024:\n\
+        add sp, 0x24\n\
+        pop {r3-r5}\n\
+        mov r8, r3\n\
+        mov r9, r4\n\
+        mov r10, r5\n\
+        pop {r4-r7}\n\
+        pop {r0}\n\
+        bx r0\n\
+        ");
+}
+#endif
+
+bool8 LoadCompressedSpriteSheetUsingHeap(const struct CompressedSpriteSheet* src)
+{
+    struct SpriteSheet dest;
+    void* buffer;
+
+    buffer = AllocZeroed(*((u32*)(&src->data[0])) >> 8);
```

Comment:
Don't need `&src->data[0]` to create a pointer to the start of `src->data`.
Just use `src->data`.

Example:
```c
    buffer = AllocZeroed(*((u32 *)src->data) >> 8);
```